internal/audit/server_checks.go

```diff
@@ -230,3 +230,48 @@ func makeCheckServerResourceLimits(memoryUsageThreshold, storeUsageThreshold flo
 		return Pass, nil
 	}
 }
+
+func checkServerLeafnodeNames() checkFunc {
+	return func(r *archive.Reader, examples *ExamplesCollection) (Outcome, error) {
+
+		checkIfNameContainsSpaces := func(name string) bool {
+			for _, c := range name {
+				if c == ' ' {
+					return true
+				}
+			}
+			return false
+		}
+
+		for _, clusterName := range r.GetClusterNames() {
+			clusterTag := archive.TagCluster(clusterName)
+
+			for _, serverName := range r.GetClusterServerNames(clusterName) {
+				serverTag := archive.TagServer(serverName)
+
+				var serverLeafz server.Leafz
+				err := r.Load(&serverLeafz, clusterTag, serverTag, archive.TagServerLeafs())
+				if err != nil {
+					return Skipped, fmt.Errorf("failed to load Leafz for server %s: %w", serverName, err)
```

Comment:
Returning here skips the entire check if one of the servers is missing `leafz`, should probably just print a warning and continue